### PR TITLE
Cli command to show registered requirements.

### PR DESCRIPTION
**Pull request by Dmitrykob**
## Description

Added new handler in the debug server that allows to get the list of requirements:
image-1

Added a new cli command for outputting the list of requirements and its description:
image-2

image-3

## Why do we need it, and what problem does it solve?

close https://github.com/deckhouse/deckhouse/issues/7709

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse-controller
type: fix
summary: Added cli command for outputting the list of requirements.
impact_level: default
```

### CODE REVIEW

File: go_lib/dependency/requirements/store.go [hunk 1]
```diff
@@ -47,3 +47,13 @@ func (m *MemoryValuesStore) Get(key string) (interface{}, bool) {
 	v, ok := m.values[key]
 	return v, ok
 }
+
+func (m *MemoryValuesStore) GetAll() interface{} {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+	var resp = make(map[string]interface{})
```
Comment (yalosev) [inline go_lib/dependency/requirements/store.go:54]:
```suggestion
	var resp = make(map[string]interface{}, len(m.values))
```

File: go_lib/dependency/requirements/store.go [hunk 2]
```diff
@@ -47,3 +47,13 @@ func (m *MemoryValuesStore) Get(key string) (interface{}, bool) {
 	v, ok := m.values[key]
 	return v, ok
 }
+
+func (m *MemoryValuesStore) GetAll() interface{} {
```
Comment (yalosev) [inline go_lib/dependency/requirements/store.go:51]: Why `interface{}` not `map[string]interface{}`

File: go_lib/dependency/requirements/store.go [hunk 3]
```diff
@@ -47,3 +47,13 @@ func (m *MemoryValuesStore) Get(key string) (interface{}, bool) {
...
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+	var resp = make(map[string]interface{})
+	for k, v := range m.values {
```
Comment (yalosev) [inline go_lib/dependency/requirements/store.go:55]: Also don't think that we need copy of the map here

File: deckhouse-controller/pkg/debug/requirements.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		}
+		defer resp.Body.Close()
+
+		var requirements = make(map[string]interface{})
+		decoder := json.NewDecoder(resp.Body)
+		if err := decoder.Decode(&requirements); err != nil {
+			return fmt.Errorf("error unmarshal requirements")
+		}
+
+		for key, value := range requirements {
```
Comment (yalosev) [inline deckhouse-controller/pkg/debug/requirements.go:42]: I guess we can yaml.Marshal to the stdout here

Comment (yalosev) [reply deckhouse-controller/pkg/debug/requirements.go:42]: Or process the `output format` option and use yaml / json

File: go_lib/dependency/requirements/store.go (hunk 1 above)
Comment (Dmitrykob) [reply go_lib/dependency/requirements/store.go:54]: Apply

File: go_lib/dependency/requirements/store.go (hunk 2 above)
Comment (Dmitrykob) [reply go_lib/dependency/requirements/store.go:51]: Apply

**Review by yalosev — APPROVED** [review]

**Pull request merged**